File: lib/src/main/java/com/sd/lib/http/RequestHandler.java
package com.sd.lib.http;

/**
 * 执行异步请求后返回的对象，提供一些方法暴露给外部
 */
public class RequestHandler
{
    private RequestTask mRequestTask;

    RequestHandler(RequestTask requestTask)
    {
        mRequestTask = requestTask;
    }

    /**
     * 取消请求
     */
    public void cancel()
    {
        mRequestTask.cancel(true);
    }
}

File: lib/src/main/java/com/sd/lib/http/RequestHandler.kt
package com.sd.lib.http

/**
 * 异步请求任务处理对象，提供一些方法暴露给外部
 */
class RequestHandler internal constructor(task: RequestTask) {
    private val _requestTask: RequestTask = task

    /**
     * 任务是否处于活动状态
     */
    val isActive: Boolean
        get() = _requestTask.isActive

    /**
     * 取消请求
     */
    fun cancel() {
        _requestTask.cancel()
    }
}